### ISSUE TITLE
Pretrained Model

### ISSUE THREAD

**Issue by bytosaur** [EleMisi/ConditionalVAE#10]
Hey EleMisi,

great work! I am looking into real-time face-manipulation using TensorFlow 2... 
Did you try using higher resolutions, say 128x128?
Do you happen to have a SavedModel, checkpoints or h5 weights for it? 
If not, how long does the training take?

many thanks :)

**Comment by EleMisi** [comment]
Hi bytosaur,

Thank you very much!

Your project sounds really interesting, here you are the saved model: [SavedModel.zip](https://github.com/EleMisi/ConditionalVAE/files/6243839/SavedModel.zip)
We tried only with 64x64 resolution, and with the following configuration the training took around 430 sec per epoch:
```
# Hyper-params
label_dim = 40
image_dim = [64, 64, 3]
latent_dim = 128
beta = 0.65

# Training configuration
learning_rate = 0.001
train_size = 0.7
n_epochs = 30
batch_size = 32
```

Let me know if I can help you, and stay safe 🍀

EleMisi

**Comment by bytosaur** [comment]
Awesome! The SavedModel is exactly what i needed!
I ll come back to you once my experiments are done :)

**Comment by EleMisi** [comment]
Perfect! ☀️

**Comment by bytosaur** [comment]
Ou looks like that's a checkpoint. I'll either try to convert to SavedModel (currently struggeling) or train it myself and export as SavedModel. Nevertheless, the code looks promising for what i have in mind :) 
I think the __ call__() function should be named call().  As described [here](https://pychao.com/2019/10/14/the-implicit-dtype-conversion-in-the-call-and-__call__-methods-of-tf-keras-model/).

**Comment by EleMisi** [comment]
Yes, we saved only the checkpoint, but the source code of our model is available [here](https://github.com/EleMisi/ConditionalVAE/blob/master/src/ConvolutionalCondVAE.py) and you can load the parameter values that I gave you by using the "Checkpoint" section of [this notebook](https://github.com/EleMisi/ConditionalVAE/blob/master/notebooks/Train_ConditionalVAE.ipynb). 
At this point it should be quite easy for you to save the model since it is a `tf.keras.Module` subclass ( as described [here](https://www.tensorflow.org/guide/saved_model#saving_a_custom_model)) 😊

**Comment by bytosaur** [comment]
Hey there!
Sorry for no response for a 2 weeks.
There were a few things which prohibited the export as a saveModel. I tried to fix those issues in this [fork](https://github.com/bytosaur/ConditionalVAE) and cleanup the model as good as i could. I am pretty sure there are still some flaws but I havent have time yet. I first need to play around with the real time face manipulations.
I'll let u know when i come across something cool. Thanks for sharing the code :)
If u find some time let me know what you think of the changes - i hope i didnt screw up the math
